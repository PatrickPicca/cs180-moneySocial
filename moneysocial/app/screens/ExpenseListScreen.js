import React, {useState, useEffect} from 'react';
import { SafeAreaView, FlatList, StyleSheet, Text, TextInput, TouchableOpacity, View, ScrollView } from 'react-native';
import colors from '../config/colors';
//import { API, graphqlOperation, Auth } from "aws-amplify";
//import * as mutations from '../../src/graphql/mutations';
//import * as queries from '../../src/graphql/queries';
//import { useFocusEffect } from '@react-navigation/native';


export default function MyComponent() {

  const GroupData = [
    {id: '1', category: 'Soda', amount: 2.99, description: "test" },
    {id: '2', category: 'Chips', amount: 1.99, description: "test" },
  ];
  const [searchQuery, setSearchQuery] = useState('');
  const [data, setData] = useState(GroupData);
  const [editingId, setEditingId] = useState(null);
  const [category, setCategory] = useState('');
  const [amount, setAmount] = useState('');
  const [description, setDescription] = useState('');

  const [user, setUser] = useState(null);

  /*const getAllUserExpenses = async () => {
    //This block of code queries a specified expense object
   // console.log("In getAllUSerExpenses handler");
    const variables = {
      filter: {
        userID : {eq: user}
      },
    };
    const newTodo = await API.graphql({ query: queries.listExpenses,  variables});
 //   console.log(newTodo.data.listExpenses.items);
    setData(newTodo.data.listExpenses.items);
  }

    useEffect(() => {
      const fetchUser = async () => {
        try {
          const authUser = await Auth.currentAuthenticatedUser({
            bypassCache: true,
          });
          const userData2 = await API.graphql(
            graphqlOperation(queries.getUser, { id: authUser.attributes.sub })
          );
          console.log(authUser.attributes.sub);
          setUser(authUser.attributes.sub);
        } catch (error) {
          console.log('Error fetching user data:', error);
        }
      }; 
      getAllUserExpenses();
      fetchUser();
    }, []);
    
    useFocusEffect(React.useCallback(() => 
    {
      getAllUserExpenses();
    }));
  // console.log(data);
  */

  const renderItem = ({ item }) => (
    <View style={styles.displayBalance}>
      {editingId === item.id ? (
        <View style={styles.editContainer}>
          <TextInput
            style={styles.editInput}
            placeholder="Enter new category"
            value={category}
            onChangeText={text => setCategory(text)}
          />
          <TextInput
            style={styles.editInput}
            placeholder="Enter new value"
            value={amount}
            onChangeText={text => setAmount(text)}
          />
          <TextInput
            style={styles.editInput}
            placeholder="Enter new description"
            value={description}
            onChangeText={text => setDescription(text)}
          />
          <TouchableOpacity
            style={styles.updateButton}
            onPress={() => handleUpdate(item.id)}
          >
            <Text style={styles.updateButtonText}>Update</Text>
          </TouchableOpacity>
        </View>
      ) : (
        <View style={styles.itemContainer}>

          <Text style={styles.displayText}>{`${item.category}: $${item.amount}`}</Text>

          <View style={styles.buttonContainer}>
            <TouchableOpacity
              style={styles.editButton}
              onPress={() => handleEdit(item)}
            >
              <Text style={styles.editButtonText}>Edit</Text>
            </TouchableOpacity>

            <TouchableOpacity
              style={styles.removeButton}
              onPress={() => handleRemove(item.id)}
            >
              <Text style={styles.removeButtonText}>Remove</Text>
            </TouchableOpacity>
          </View>
          
        </View>
      )}
    </View>
  );

  const handleEdit = (item) => {
    setEditingId(item.id);
    setCategory(item.category);
    setDescription(item.description);
    setAmount(item.amount.toString());
  };

  /*const handleUpdate =async (id) => {

    //Need to log the id to see if its represents the expenses's unique id.
    //Need to run update query based on current values stored for that expense.

    const variables2 = {
      id : id,
      amount : amount,
      description : description,
      category : category
    };
    const newTodo2 = await API.graphql({ query: mutations.updateExpense,  variables: { input: variables2 }});
    getAllUserExpenses();
    /*
    const updatedData = data.map((item) => {
      if (item.id === id) {
        return { ...item, category, amount: parseFloat(amount) };
      }
      return item;
    });
    setData(updatedData);
    
    setEditingId(null);
    setCategory('');
    setAmount('');
    setDesc('');
  };
*/
  /*const handleRemove = async (id) => {
    const theID = id;
    console.log("id in question: " + id);
    const variables = {
      input: {
        id : theID
      },
    };
    const newTodo = await API.graphql({ query: mutations.deleteExpense,  variables});
    //const updatedData = data.filter(item => item.id !== id);

    //Instead of defining updatedData, we simply will use a mutation to delete that expense.
    getAllUserExpenses();

    //setData(updatedData);
  };
*/
  const filteredData = data.filter(item => item.category.toLowerCase().includes(searchQuery.toLowerCase()))
  
  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.label}>Expense List:</Text>
      <View style={styles.searchContainer}>
        <TextInput
          style={styles.searchInput}
          placeholder="Filter expenses"
          value={searchQuery}
          onChangeText={query => setSearchQuery(query)}
        />
      </View>
      <FlatList
        data={filteredData}
        renderItem={renderItem}
        keyExtractor={(item) => item.id}
      />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.background,
  },
  buttonContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  displayBalance: {
    padding: 20,
    borderBottomWidth: 1,
    borderBottomColor: 'gray',
  },
  itemContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  displayText: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  editContainer: {
    paddingHorizontal: 10,
    paddingVertical: 5,
    alignItems: 'center'
  },
  editButton: {
    backgroundColor: 'green',
    padding: 10,
    borderRadius: 5,
    marginRight: 10,
  },
  editButtonText: {
    color: 'white',
    fontWeight: 'bold',
  },
  editInput: {
    backgroundColor: 'white',
    paddingHorizontal: 10,
    paddingVertical: 5,
    borderRadius: 20,
    borderWidth: 1,
    borderColor: 'gray',
    marginBottom: 10,
  },
  searchContainer: {
    paddingHorizontal: 20,
    paddingVertical: 10,
  },
  searchInput: {
    backgroundColor: 'white',
    paddingHorizontal: 10,
    paddingVertical: 5,
    borderRadius: 20,
    borderWidth: 1,
    borderColor: 'gray',
  },
  label: {
    fontSize: 24,
    fontWeight: 'bold',
    marginVertical: 10,
    marginHorizontal: 20,
    paddingTop: 20,
  },
  editContainer: {
    paddingVertical: 15,
    borderBottomWidth: 1,
    borderBottomColor: colors.background,
  },
  updateButton: {
    backgroundColor: 'green',
    paddingTop: 2,
    paddingVertical: 5,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: 'gray',
    color: 'white',
    textAlign: 'center',
    alignItems: 'center'
  },
  updateButtonText: {
    color: 'white',
    fontWeight: 'bold',
  },
  removeButtonText: {
    color: 'white',
    fontWeight: 'bold',
  },
  removeButton: {
    backgroundColor: 'red',
    padding: 10,
    borderRadius: 5,
  },
})
