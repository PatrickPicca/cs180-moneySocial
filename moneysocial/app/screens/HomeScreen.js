import React, {useState} from 'react';
import { StyleSheet, Text, View, Dimensions, TextInput, Pressable } from "react-native";
import Svg, {Image, Ellipse, ClipPath} from "react-native-svg";
import Animated, {useSharedValue, useAnimatedStyle, interpolate, withTiming, withDelay} from 'react-native-reanimated';

import colors from '../config/colors'
import { useNavigation, useNavigationState } from '@react-navigation/native';
//import stackNavigator from '../Routes/MainNavigation';
//import WelcomeScreen from './WelcomeScreen';

const {width, height} = Dimensions.get('window');

function PersonalExpenseScreen(props) {
    const {height, width} = Dimensions.get('window');
    const imagePosition = useSharedValue(1);
    const [isRegistering, setIsRegistering] = useState(false);

    const myValue = 50;

    //Here we will need the entire list of Expense objects, of all categories, from this specific user.
      //In the view section you should be able to view the following
          //Current overall total expense for the past month.
            //This can be display with a number as well as a pie chart breakdown oc categories by color.
          //Button option to view total expense for the past month by category.
          //Button option to view the list of all individual expenses of all time.
          //Button option to make an expense.
            //This should take you to some Expense screen where you can fill out a varying list of details for that one expense.
    
    //We will also need a button that lead to the WelcomeScreen that logs you out.
    //We will also need to add a group view button, that would take you to the GroupExpense Screen.


    const imageAnimatedStyle = useAnimatedStyle(() => {
        const interpolation = interpolate(imagePosition.value, [0,1], [-height/2, 0])
        return {
            transform: [{translateY: withTiming(interpolation, {duration: 1000})}]
        };
    });

    const buttonsAnimatedStyle = useAnimatedStyle (() =>{
        const interpolation = interpolate(imagePosition.value, [0,1], [250,0])
        return {
            opacity: withTiming(imagePosition.value, 500),
            transform: [{translateY: withTiming(interpolation, {duration: 1000})}]
        }
    })

    const closeButtonContainerStyle = useAnimatedStyle (() =>{
        const interpolation = interpolate(imagePosition.value, [0,1], [180,360])
        return {
            opacity: withTiming(imagePosition.value === 1 ? 0 : 1, {duration: 800}),
            transform: [{rotate: withTiming(interpolation + "deg", {duration: 1000})}]
        }
    })

    const formAnimatedStyle = useAnimatedStyle(() => {
        const interpolation = interpolate(imagePosition.value, [0,1], [-height/2 + 35,0])
        return {
            opacity: imagePosition.value === 0 ? withDelay(400, withTiming(1,{duration: 1000})) : withTiming(0, {duration: 300}),
            transform: [{translateY: withTiming(interpolation, {duration: 1000})}]
        }
    })

    const loginHandler = () => {
        imagePosition.value = 0;
        if(isRegistering){
            setIsRegistering(false);
        }
    }
    const registerHandler = () => {
        imagePosition.value = 0;
        if(!isRegistering){
            setIsRegistering(true);
        }
    }

    const WelcomeScreenHandler = () => {
      props.navigation.navigate('WelcomeScreen');
    }

    const GroupScreenHandler = () => {
      props.navigation.navigate('groupList');
    }

    const ExpenseListScreenHander = () => {
      props.navigation.navigate('ExpenseListScreen');
    }

    return (

      <View style={styles.container}>

        

        <Animated.View style={[StyleSheet.absoluteFill, imageAnimatedStyle]}>
          <View style={styles.displayBalance}>
            <Text style={styles.displayText}>{'Monthly Expenses: $' + myValue}</Text>
          </View>
        </Animated.View>
      </View>
      
    );

    /*
    <View style={styles.container}>
      <Animated.View style={[StyleSheet.absoluteFill, imageAnimatedStyle]}>
        <Svg height={height} width={width}>
            <ClipPath id = "clipPathId">
                <Ellipse cx={width/2} rx={height} ry={height}/>
            </ClipPath>
          <Image
            href={require("../assets/moneysocial-logo.png")}
            width={width}
            height = {height-250}
            clipPath = "url(#clipPathId)"
          />
        </Svg>
        <Animated.View style={[styles.closeButtonContainer, closeButtonContainerStyle]}>
        <Text onPress={() => (imagePosition.value = 1)}>X</Text>
        </Animated.View>
      </Animated.View>
      <View style={styles.bottomContainer}>
       <Animated.View style={buttonsAnimatedStyle}>
        <Pressable style={styles.button} onPress={loginHandler}>
          <Text style={styles.buttonText}>LOG IN</Text>
        </Pressable>
       </Animated.View>
       <Animated.View style={buttonsAnimatedStyle}>
        <Pressable style={styles.button} onPress={registerHandler}>
          <Text style={styles.buttonText}>REGISTER</Text>
        </Pressable>
       </Animated.View>
        <Animated.View style={[styles.formInputContainer, formAnimatedStyle]}>
          <TextInput
            placeholder="Email"
            placeholderTextColor="black"
            style={styles.textInput}
          />
          {isRegistering && (
            <TextInput
            placeholder="Full Name"
            placeholderTextColor="black"
            style={styles.textInput}
          />
          )}
          <TextInput
            placeholder="Password"
            placeholderTextColor="black"
            style={styles.textInput}
            /> 
          <View style={styles.formButton}>
            <Text style={styles.buttonText}>{isRegistering ? 'REGISTER' : 'LOG IN'}</Text>
          </View>
        </Animated.View>
      </View>
    </View>

    );
    */
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'flex-end',
        backgroundColor: colors.background,
      },
      button: {
        
        backgroundColor: colors.primary,
        height: 55,
        width: 60,
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 35,
        marginHorizontal: 5,
        marginVertical: 10,
        borderWidth: 2,
        borderColor: 'white'
      },
      bottombutton: {
        
        backgroundColor: colors.primary,
        height: 55,
        width: 60,
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 35,
        marginHorizontal: 5,
        marginVertical: 10,
        borderWidth: 2,
        borderColor: 'pink'
      },
      buttonText: {
        fontSize: 20,
        fontWeight: '600',
        color: 'white',
        letterSpacing: 0.5
      },
      bottombuttonText: {
        fontSize: 15,
        fontWeight: '600',
        color: 'white',
        letterSpacing: 0.5
      },
      displayText: {
        fontSize: 20,
        fontWeight: '600',
        color: 'white',
        letterSpacing: 0.5
      },
      bottomContainer: {
        justifyContent: 'center',
        height: height / 8,
      },
      bottomScreenHeader: {
        flexDirection: 'row',
        justifyContent: 'center',
        height: height / 8,
      },
      textInput: {
        height: 50,
        borderWidth: 1,
        borderColor: colors.primary,
        marginHorizontal: 20,
        marginVertical: 10,
        borderRadius: 25,
        paddingLeft: 10
      },
      formButton: {
        backgroundColor: colors.primary,
        height: 55,
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 35,
        marginHorizontal: 20,
        marginVertical: 10,
        borderWidth: 1,
        borderColor: 'white',
        shadowColor: "#000",
        shadowOffset: {
          width: 0,
          height: 4,
        },
        
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
      },
      displayBalance: {
        backgroundColor: colors.accent,
        height: 55,
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 25, //Affects the radius of the corners
        marginHorizontal: 20,
        marginVertical: 10,
        borderWidth: 1,
        borderColor: 'white',
        shadowColor: "#000",
        shadowOffset: {
          width: 0,
          height: 4,
        },
        top: +50,
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
      },
      formInputContainer: {
        marginBottom: 70,
        ...StyleSheet.absoluteFill,
        zIndex: -1,
        justifyContent: 'center'
      },
      closeButtonContainer: {
        height: 40,
        width: 40,
        justifyContent: 'center',
        alignSelf: 'center',
        shadowColor: "#000",
        shadowOffset: {
          width: 0,
          height: 5,
        },
        shadowOpacity: 0.34,
        shadowRadius: 6.27,
        elevation: 1,
        backgroundColor: 'white',
        alignItems: 'center',
        borderRadius: 20,
        top: -25
    }
})

export default PersonalExpenseScreen;