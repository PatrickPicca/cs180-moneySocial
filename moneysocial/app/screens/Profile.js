import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { PieChart } from 'react-native-chart-kit';
import colors from '../config/colors';

const MyComponent = () => {
  const data = [
    {
      name: 'Expense 1',
      amount: 100,
      color: colors.primary,
    },
    {
      name: 'Expense 2',
      amount: 200,
      color: colors.secondary,
    },
    {
      name: 'Expense 3',
      amount: 150,
      color: colors.tertiary,
    },
  ];

  const chartConfig = {
    backgroundColor: colors.background,
    backgroundGradientFrom: colors.background,
    backgroundGradientTo: colors.background,
    color: (opacity = 1) => `rgba(255, 255, 255, ${opacity})`,
    strokeWidth: 2, // optional, default 3
    barPercentage: 0.5,
    useShadowColorFromDataset: false // optional
  };

  const chartData = data.map((item) => ({
    name: item.name,
    amount: item.amount,
    color: item.color,
    legendFontColor: '#7F7F7F',
    legendFontSize: 10,
  }));

  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.label}>Expense Graphs</Text>
      <View style={styles.chartContainer}>
        <PieChart
          data={chartData}
          width={300}
          height={175}
          chartConfig={chartConfig}
          accessor="amount"
          backgroundColor="transparent"
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.background,
  },
  label: {
    fontSize: 24,
    fontWeight: 'bold',
    marginVertical: 10,
    marginHorizontal: 20,
    paddingTop: 20,
    paddingBottom: 10
  },
  chartContainer: {
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default MyComponent;
