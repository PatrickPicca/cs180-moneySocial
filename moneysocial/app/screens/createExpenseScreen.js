import React, { useState } from 'react';
import {StyleSheet, Text, View, Dimensions, TextInput, Pressable, TouchableOpacity} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import Ionicons from 'react-native-vector-icons/Ionicons';
import colors from '../config/colors';

function CreateExpenseScreen() {
  const navigation = useNavigation();

  const [title, setTitle] = useState('');
  const [desc, setDesc] = useState('');
  const [priority, setPriority] = useState('');

  const handleCreateExpense = () => {
    navigation.pop();
  };

  return (
    <View style={styles.container}>
      <Text style={styles.label}>Add Expense:</Text>
      <TextInput
        style={styles.input}
        placeholder="Title"
        value={title}
        onChangeText={(value) => setTitle(value)}
      />
      <TextInput
        style={styles.input}
        placeholder="Description"
        multiline
        value={desc}
        onChangeText={(value) => setDesc(value)}
      />
      <TextInput
        style={styles.input}
        placeholder="Set Priority"
        keyboardType="number-pad"
        value={priority}
        onChangeText={(value) => setPriority(value)}
      />

      <TouchableOpacity style={styles.returnbutton} onPress={handleCreateExpense}>
        <Ionicons name={'arrow-back-outline'} size={40} />
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: colors.background,
      justifyContent: 'center',
      paddingHorizontal: 20,
    },
    input: {
      width: '100%',
      borderWidth: 1,
      borderColor: '#C0C0C0',
      borderRadius: 20,
      backgroundColor: '#fff',
      fontSize: 20,
      marginVertical: 10,
      paddingHorizontal: 15,
      paddingVertical: 5,
    },
    button: {
      backgroundColor: colors.primary,
      height: 55,
      width: 60,
      alignItems: 'center',
      justifyContent: 'center',
      borderRadius: 35,
      marginVertical: 10,
      borderWidth: 2,
      borderColor: 'white',
      alignSelf: 'center',
    },
    returnbutton: {
      backgroundColor: colors.primary,
      height: 50,
      width: 70,
      alignItems: 'center',
      justifyContent: 'center',
      borderRadius: 70,
      marginVertical: 10,
      borderWidth: 2,
      borderColor: 'white',
      alignSelf: 'center'
    },
    label: {
      fontSize: 24,
      fontWeight: 'bold',
      marginVertical: 10,
      marginHorizontal: 20,
    },
  });
  

export default CreateExpenseScreen;
