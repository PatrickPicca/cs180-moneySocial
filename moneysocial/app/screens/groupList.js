import React, {useState} from 'react';
import { StyleSheet, Text, View, Dimensions, TextInput, Pressable, FlatList, ScrollView } from "react-native";
import Svg, {Image, Ellipse, ClipPath} from "react-native-svg";
import Animated, {useSharedValue, useAnimatedStyle, interpolate, withTiming, withDelay} from 'react-native-reanimated';

import colors from '../config/colors'



const {width, height} = Dimensions.get('window');

function PersonalExpenseScreen(props) {
    const {height, width} = Dimensions.get('window');
    const imagePosition = useSharedValue(1);
    const [isRegistering, setIsRegistering] = useState(false);
    
    const myValue1 = 360;
    const myValue2 = 1050;


    
    //Here we will list all groups you are a part of, alongside any current, total expenses you have made for that group.
      //In the view section you should be able to view / do the following
          //A list of all groups you are in and the current total expenses for them.
            //A button option to enter that group to view expenses for the past option.
            //A button option to expenses by a certain category for the past month.
            //A button option to make an expense.
          //Button option to view total expense for the past month by category.
          //Button option to view the list of all individual expenses of all time.
            //This should take you to some Expense screen where you can fill out a varying list of details for that one expense.
          //Button option to exit the current group.
          //Display somewhere, probably the header, a code unique to that group that others need to join.

    
    //Need a button that exits the groupExpense screen, taking you to you to the PersonalExpense screen.
    //Need some join group prompt with textbox where a user can type in a unique group code.
      //On success that group gets added to the user's list of active groups.
      //On failure, no group is joined.


    const imageAnimatedStyle = useAnimatedStyle(() => {
        const interpolation = interpolate(imagePosition.value, [0,1], [-height/2, 0])
        return {
            transform: [{translateY: withTiming(interpolation, {duration: 1000})}]
        };
    });

    const buttonsAnimatedStyle = useAnimatedStyle (() =>{
        const interpolation = interpolate(imagePosition.value, [0,1], [250,0])
        return {
            opacity: withTiming(imagePosition.value, 500),
            transform: [{translateY: withTiming(interpolation, {duration: 1000})}]
        }
    })

    const closeButtonContainerStyle = useAnimatedStyle (() =>{
        const interpolation = interpolate(imagePosition.value, [0,1], [180,360])
        return {
            opacity: withTiming(imagePosition.value === 1 ? 0 : 1, {duration: 800}),
            transform: [{rotate: withTiming(interpolation + "deg", {duration: 1000})}]
        }
    })

    const formAnimatedStyle = useAnimatedStyle(() => {
        const interpolation = interpolate(imagePosition.value, [0,1], [-height/2 + 35,0])
        return {
            opacity: imagePosition.value === 0 ? withDelay(400, withTiming(1,{duration: 1000})) : withTiming(0, {duration: 300}),
            transform: [{translateY: withTiming(interpolation, {duration: 1000})}]
        }
    })

    const WelcomeScreenHandler = () => {
    //  props.navigation.navigate('WelcomeScreen');
    }
    const groupListHandler = () => {
      console.log("logging in");
      }

    const ExpenseListScreenHander = () => {
      props.navigation.navigate('ExpenseListScreen');
    }
    const HomeScreenHandler = () => {
      props.navigation.navigate('HomeScreen');
    }




    const DATA = [
      {
        id: 'bd7acbea-c1b1-46c2-aed5-3ad53abb28ba',
        title: 'First Item',
      },
      {
        id: '3ac68afc-c605-48d3-a4f8-fbd91aa97f63',
        title: 'Second Item',
      },
      {
        id: '58694a0f-3da1-471f-bd96-145571e29d72',
        title: 'Third Item',
      },
      {
        id: '5869d72',
        title: '6th Item',
      },
    ];

    const Item = ({title}) => (
      <View style={styles.item}>
        <Text style={styles.title}>{title}</Text>
        <Pressable style={styles.bottombutton} onPress={groupListHandler}>
          <Text style={styles.bottombuttonText}>Logout</Text>
        </Pressable>

      </View>
    );

    //View should have a list of all groups the user is in, as well as that groups current total expense.
    return (
      
      <View style={{flex:1}}>onPress
        <FlatList
          data={DATA}
          renderItem={({item}) => <Item title={item.title} />}
          keyExtractor={item => item.id}
        />
        <View style={styles.container}>

          <Animated.View style={buttonsAnimatedStyle}>
            <View style={styles.bottomScreenHeader}>
              <Pressable style={styles.bottombutton} onPress={WelcomeScreenHandler}>
                <Text style={styles.bottombuttonText}>Logout</Text>
              </Pressable>

              <Pressable style={styles.bottombutton} onPress={HomeScreenHandler}>
                <Text style={styles.bottombuttonText}>Home</Text>
              </Pressable>

              <Pressable style={styles.bottombutton} onPress={ExpenseListScreenHander}>
                <Text style={styles.bottombuttonText}>Details</Text>
              </Pressable>

              
              
            </View>  
          </Animated.View>
        </View>
      </View>
    );
    
}

/* 
          <ScrollView style={styles.scrollView}>
            <View style={styles.topContainer}>

              <View style={styles.displayBalance}>
                <Text style={styles.displayText}>{'Group 1: $' + myValue1}</Text>
              </View>
              <View style={styles.displayBalance}>
                <Text style={styles.displayText}>{'Group 2: $' + myValue2}</Text>
              </View>
            </View>
          </ScrollView>
          <Animated.View style={[StyleSheet.absoluteFill, imageAnimatedStyle]}>

          </Animated.View> 
*/

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'flex-end',
        backgroundColor: colors.background,
      },
      button: {
        
        backgroundColor: colors.primary,
        height: 55,
        width: 60,
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 35,
        marginHorizontal: 5,
        marginVertical: 10,
        borderWidth: 2,
        borderColor: 'white'
      },
      bottombutton: {
        
        backgroundColor: colors.primary,
        height: 55,
        width: 60,
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 35,
        marginHorizontal: 5,
        marginVertical: 10,
        borderWidth: 2,
        borderColor: 'pink'
      },
      buttonText: {
        fontSize: 20,
        fontWeight: '600',
        color: 'white',
        letterSpacing: 0.5
      },
      bottombuttonText: {
        fontSize: 15,
        fontWeight: '600',
        color: 'white',
        letterSpacing: 0.5
      },
      bottomContainer: {
        justifyContent: 'center',
        height: height / 2,
      },
      bottomScreenHeader: {
        flexDirection: 'row',
        justifyContent: 'center',
        height: height / 8,
      },
      textInput: {
        height: 50,
        borderWidth: 1,
        borderColor: colors.primary,
        marginHorizontal: 20,
        marginVertical: 10,
        borderRadius: 25,
        paddingLeft: 10
      },
      formButton: {
        backgroundColor: colors.primary,
        height: 55,
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 35,
        marginHorizontal: 20,
        marginVertical: 10,
        borderWidth: 1,
        borderColor: 'white',
        shadowColor: "#000",
        shadowOffset: {
          width: 0,
          height: 4,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
      },
      topContainer: {
        justifyContent: 'center',
        height: height / 4,
        top: 50,
      },
      formInputContainer: {
        marginBottom: 70,
        ...StyleSheet.absoluteFill,
        zIndex: -1,
        justifyContent: 'center'
      },
      closeButtonContainer: {
        height: 40,
        width: 40,
        justifyContent: 'center',
        alignSelf: 'center',
        shadowColor: "#000",
        shadowOffset: {
          width: 0,
          height: 5,
        },
        shadowOpacity: 0.34,
        shadowRadius: 6.27,
        elevation: 1,
        backgroundColor: 'white',
        alignItems: 'center',
        borderRadius: 20,
        top: -25
    }
})

export default PersonalExpenseScreen;