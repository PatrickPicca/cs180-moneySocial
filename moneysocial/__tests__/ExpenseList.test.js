import React from 'react';
import { render } from '@testing-library/react-native';
import ExpenseList from "../app/screens/ExpenseListScreen";
import renderer from 'react-test-renderer';
import { renderItem } from '../app/screens/ExpenseListScreen';

describe('Expense List component', () => {
    it('renders correctly', () => {
      render(<ExpenseList id='1' />);
    });
    
  });

test('renders correctly', () => {
  const item = { id: 1, category: 'Food', amount: 10.0, description: 'Groceries' };
  const editingId = null;
  const category = '';
  const amount = '';
  const description = '';
  const handleEdit = jest.fn();
  const handleRemove = jest.fn();
  const handleUpdate = jest.fn();

  const tree = renderer.create(
    <renderItem
      item={item}
      editingId={editingId}
      category={category}
      amount={amount}
      description={description}
      handleEdit={handleEdit}
      handleRemove={handleRemove}
      handleUpdate={handleUpdate}
    />
  ).toJSON();
  
  expect(tree).toMatchSnapshot();
});


  