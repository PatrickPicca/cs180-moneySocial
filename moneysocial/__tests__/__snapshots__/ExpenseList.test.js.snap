// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<renderItem
  amount=""
  category=""
  description=""
  editingId={null}
  handleEdit={[MockFunction]}
  handleRemove={[MockFunction]}
  handleUpdate={[MockFunction]}
  item={
    Object {
      "amount": 10,
      "category": "Food",
      "description": "Groceries",
      "id": 1,
    }
  }
/>
`;
